### PR TITLE
Add packages to execute functions inside SQL transactions

### PR DESCRIPTION
**Pull request by nicolaasuni**

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-1.18.10
```
Comment (BoRuDar) [inline VERSION:1]: Could we use git tags instead of the file?

File: pkg/sqltransaction/sqltransaction.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+package sqltransaction
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+
+	"github.com/nexmoinc/gosrvlib/pkg/logging"
+	"go.uber.org/zap"
```
Comment (BoRuDar) [inline pkg/sqltransaction/sqltransaction.go:11]: Are we using 2 logging libs at the same time?

File: pkg/sqlxtransaction/sqlxtransaction.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+// Exec executes the specified function inside a SQL transaction.
+func Exec(ctx context.Context, db *sqlx.DB, run ExecFunc) error {
+	tx, err := db.BeginTxx(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("unable to start an SQLX transaction: %w", err)
+	}
+
+	defer func() {
+		if err := tx.Rollback(); err != nil && !errors.Is(err, sql.ErrTxDone) {
```
Comment (BoRuDar) [inline pkg/sqlxtransaction/sqlxtransaction.go:26]: Is it safe to call `tx.Rollback()` after `tx.Commit()` finished successfully? It seems logically incorrect to do so.

**Review by BoRuDar — CHANGES_REQUESTED** [review]

File: pkg/sqlxtransaction/sqlxtransaction.go (hunk 3 above)
Comment (mgiaccone) [reply pkg/sqlxtransaction/sqlxtransaction.go:26]: I think that is safe to do

File: pkg/sqltransaction/sqltransaction.go (hunk 2 above)
Comment (mgiaccone) [reply pkg/sqltransaction/sqltransaction.go:11]: zap is used for the `zap.Error`. `logging` is mostly for context/initialisation

File: VERSION (hunk 1 above)
Comment (mgiaccone) [reply VERSION:1]: That would be brilliant, there is no real need for a version file

Comment (nicolaasuni) [reply VERSION:1]: We use both.

File: pkg/sqltransaction/sqltransaction.go (hunk 2 above)
Comment (nicolaasuni) [reply pkg/sqltransaction/sqltransaction.go:11]: No.

File: pkg/sqlxtransaction/sqlxtransaction.go (hunk 3 above)
Comment (nicolaasuni) [reply pkg/sqlxtransaction/sqlxtransaction.go:26]: Yes as we check for the sql.ErrTxDone.

File: VERSION (hunk 1 above)
Comment (nicolaasuni) [reply VERSION:1]: The VERSION file is the SSOT for the version and it is used for CI/CD and tagging.

**Review by BoRuDar — APPROVED** [review]

**Pull request merged**